src/core_read.cpp

```diff
@@ -117,23 +117,23 @@ bool DecodeHexTx(CMutableTransaction& tx, const std::string& hex_tx, bool try_no
 
     std::vector<unsigned char> txData(ParseHex(hex_tx));
 
-    if (try_no_witness) {
-        CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS);
+    if (try_witness) {
+        CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION);
         try {
             ssData >> tx;
-            if (ssData.eof() && (!try_witness || CheckTxScriptsSanity(tx))) {
+            if (ssData.empty()) {
                 return true;
             }
         } catch (const std::exception&) {
             // Fall through.
         }
     }
 
-    if (try_witness) {
-        CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION);
+    if (try_no_witness) {
+        CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS);
         try {
             ssData >> tx;
-            if (ssData.empty()) {
+            if (ssData.eof() && (!try_witness || CheckTxScriptsSanity(tx))) {
```

Comment:
No, in the other clause. It's supposed to bypass the sanity check when only one of try_witness and try_no_witness is set.

Reply:
ah I see, fixed it up(within my understanding) and put a quick comment there for future readers' comprehension